### PR TITLE
feat(datetime-input): add component #IX-3628

### PR DESCRIPTION
**Pull request by alexkaduk**
## 💡 What is the current behavior?

No datetime input component exists. Users must use separate date and time inputs.

## 🆕 What is the new behavior?

Added `ix-datetime-input` component - a combined date and time input with integrated picker.

**Features:**
- Combines date and time selection in single input
- Wraps `ix-datetime-picker` with auto-open/close behavior
- Display format: `dateFormat + timeFormat`
- Full validation: format, min/max date, required

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [ ] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📕 Add or update a Storybook story
- [ ] 📄 Documentation was reviewed/updated [siemens/ix-docs](https://github.com/siemens/ix-docs)
- [ ] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [ ] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [ ] 🧐 Static code analysis passes (`pnpm lint`)
- [ ] 🏗️ Successful compilation (`pnpm build`, changes pushed)

Related docu PR is https://github.com/siemens/ix-docs/pull/165

### CODE REVIEW

File: packages/vue-test-app/src/Root.vue [hunk 1]
```diff
@@ -480,4 +494,3 @@ const currentView = computed(() => {
     <component :is="currentView"></component>
   </IxApplicationContext>
 </template>
-import layoutAutoVue from './preview-examples/layout-auto.vue';
```
Comment (nuke-ellington) [inline packages/vue-test-app/src/Root.vue:483]: Should this not be included somewhere?

File: packages/html-test-app/src/preview-examples/quick-test.js
Comment (nuke-ellington) [inline packages/html-test-app/src/preview-examples/quick-test.js:1]: Can be purged?

File: packages/core/src/components/date-time-card/date-time-card.tsx [hunk 2]
```diff
@@ -40,6 +40,11 @@ export class DateTimeCard {
    */
   @Prop() corners: DateTimeCardCorners = 'rounded';
 
+  /**
+   * Remove content padding
+   */
+  @Prop() noPadding: boolean = false;
```
Comment (nuke-ellington) [inline packages/core/src/components/date-time-card/date-time-card.tsx:46]: Since tag missing

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+  /** Placeholder text when input is empty */
+  @Prop({ reflect: true }) placeholder?: string;
+
+  /** Value in display format (e.g., "2026/01/21 13:07:04" for default dateFormat + timeFormat) */
+  @Prop({ reflect: true, mutable: true }) value?: string = '';
+
+  /** Luxon date format for display (e.g., 'yyyy/LL/dd' → "2026/01/20") */
+  @Prop() dateFormat: string = 'yyyy/LL/dd';
+
+  /** Luxon time format for display (e.g., 'HH:mm:ss' → "13:07:04") */
```
Comment (nuke-ellington) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: In the other date and time components we have linked to the luxon docs for available patterns. We should do so here, too.

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+  @Prop() i18nTime: string = 'Time';
+
+  /** ARIA label for previous month navigation button */
+  @Prop() ariaLabelPreviousMonthButton?: string;
+
+  /** ARIA label for next month navigation button */
+  @Prop() ariaLabelNextMonthButton?: string;
+
+  /** ARIA label for the calendar icon button */
+  @Prop() ariaLabelCalendarButton?: string;
```
Comment (nuke-ellington) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: We should set english defaults (e.g. "Previous month")

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 5]
```diff
@@ -0,0 +1,643 @@
...
+    if (!this.value) {
+      const now = DateTime.now();
+      if (now.isValid) {
+        this.from = now.toFormat(this.dateFormat);
+        this.time = now.toFormat(this.timeFormat);
+      }
+    }
+  }
+
+  private readonly onCalendarClick = (event: Event) => {
```
Comment (nuke-ellington) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: Clickable icon is a11y antipattern AFAIK.
TBD needs to be aligned with new keynav concept

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 6]
```diff
@@ -0,0 +1,643 @@
...
+          slotEndRef={this.slotEndRef}
+          onSlotChange={() => this.updatePaddings()}
+        >
+          <ix-icon-button
+            aria-expanded={this.show}
+            aria-label={
+              this.ariaLabelCalendarButton || 'Toggle datetime picker'
+            }
+            class={{ 'calendar-hidden': this.disabled || this.readonly }}
+            data-testid="open-datetime-picker"
```
Comment (nuke-ellington) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: Select via aria-label instead in tests
```suggestion
```

File: packages/vue-test-app/src/Root.vue (hunk 1 above)
Comment (alexkaduk) [reply packages/vue-test-app/src/Root.vue:483]: we have it here
`import LayoutAuto from './preview-examples/layout-auto.vue';`
https://github.com/siemens/ix/blob/944c86d1fc848171d0a80dc02b53fb82ceb21fcd/packages/vue-test-app/src/Root.vue#L153C1-L153C61

File: packages/html-test-app/src/preview-examples/quick-test.js
Comment (alexkaduk) [reply packages/html-test-app/src/preview-examples/quick-test.js:1]: my bad. deleted

File: packages/core/src/components/date-time-card/date-time-card.tsx (hunk 2 above)
Comment (alexkaduk) [reply packages/core/src/components/date-time-card/date-time-card.tsx:46]: added. thanks

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 3 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: added. thanks

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 5 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: will discuss on dev sync

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 6 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: done. thanks

File: packages/core/src/components/datetime-input/test/datetime-input.ct.ts [hunk 7]
```diff
@@ -0,0 +1,605 @@
...
+        .click();
+    },
+    confirm: async () => {
+      const confirmButton = dateTimeInput.locator(
+        'ix-datetime-picker .btn-select-date'
+      );
+      await confirmButton.click();
+    },
+  };
+};
```
Comment (danielleroux) [inline packages/core/src/components/datetime-input/test/datetime-input.ct.ts]: Remove all css selectors (e.g chosen attribute selectors) use aria attributes to select e.g getByRole, getLabel etc

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 8]
```diff
@@ -0,0 +1,643 @@
...
+  createValidityState,
+  handleIconClick,
+  openDropdown as openDropdownUtil,
+} from '../utils/input/picker-input.util';
+import { makeRef } from '../utils/make-ref';
+import { DateTimeInputValidityState } from './datetime-input.types';
+
+/**
+ * @form-ready
+ *
```
Comment (danielleroux) [inline packages/core/src/components/datetime-input/datetime-input.tsx:51]: Since annotation is missing (4.4.0)

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 6 above)
Comment (danielleroux) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: @nuke-ellington @alexkaduk icon-button can be complete set to aria-hidden="true" and tab-index=-1 because the dropdown should be open via focus of the input + arrow keys. This is currently not that easy possible to implement due some padding refactoring of https://github.com/siemens/ix/pull/2268.

I would suggest not to introduce any new ariaLabelCalenderButton property due the fact that is not necessary any more later on. Stick with the default text "Toggle datetime picker". @nuke-ellington  any thoughts?

**Comment by alexkaduk** [comment]
> Why did we change all of the examples to February 8th 2026

PO requirement

> When deleting charactes manually in input while dropdown is open focus gets lost disrupting the edit

updated. thanks. appreciate checking if it works as expected now

> Storybook story missing

added, thanks

File: packages/core/src/components/datetime-input/test/datetime-input.ct.ts (hunk 7 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/test/datetime-input.ct.ts]: removed where it is possible. pls check. in case there are parts that might be adjusted, pls suggest

**Comment by alexkaduk** [comment]
> Value accessor selector needs to be adapted for angular (ngModule)

added, thanks

> Visual regression test is missing which shows an input with an open dropdown, maybe not all cases are needed but some edge cases would be nice

added test case with opened dropdown

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 8 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx:51]: added. thanks

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 6 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: What was done based on comments for the calendar icon button in datetime-input.tsx:

Set aria-hidden="true"
Set tabindex="-1"
Removed the ariaLabelCalendarButton prop entirely
Used hardcoded aria-label="Toggle datetime picker"
Removed data-testid="open-datetime-picker" from the button
Updated all tests to use CSS selector .locator('ix-icon-button').first() for this button

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 4 above)
Comment (nuke-ellington) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: @alexkaduk - this is still relevenat for `ariaLabelPreviousMonthButton` and `ariaLabelNextMonthButton`.

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 9]
```diff
@@ -0,0 +1,683 @@
...
+  @State() isInvalid: boolean = false;
+
+  /** Validation state: Valid */
+  @State() isValid: boolean = false;
+
+  /** Validation state: Info */
+  @State() isInfo: boolean = false;
+
+  /** Validation state: Warning */
+  @State() isWarning: boolean = false;
```
Comment (danielleroux) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: Why do you added this comments there are super self explaining states. They dont need any comment annotation

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 10]
```diff
@@ -0,0 +1,683 @@
...
+  /** Prevent form submission when Enter is pressed */
+  @Prop({ reflect: true }) suppressSubmitOnEnter: boolean = false;
+
+  /** Text alignment within the input field */
+  @Prop() textAlignment: 'start' | 'end' = 'start';
+
+  /**
+   * Enable Popover API rendering for dropdown.
+   *
+   * @default false
```
Comment (danielleroux) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: `@default` can be removed because api documentation will show the default based on the typing

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 11]
```diff
@@ -0,0 +1,683 @@
...
+  private readonly datetimePickerRef = makeRef<HTMLIxDatetimePickerElement>();
+
+  /** Dropdown open/closed state */
+  @State() show: boolean = false;
+
+  /** Date value for picker (in picker's format) */
+  @State() from?: string | null = null;
+
+  /** Time value for picker (in picker's format) */
+  @State() time?: string | null = null;
```
Comment (danielleroux) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: Why do you added this comments there are super self explaining states. They dont need any comment annotation

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 12]
```diff
@@ -0,0 +1,683 @@
...
+
+  private classObserver?: ClassMutationObserver;
+  /** @internal */
+  public invalidReason?: string;
+  /** @internal */
+  public touched = false;
+  /** @internal */
+  public validityTracker: PickerValidityStateTracker =
```
Comment (danielleroux) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: Why do you added this jsdocs to each property? It does not need any internal js docs annotation because its the internal state

File: packages/core/src/components/datetime-input/datetime-input.types.ts
Comment (danielleroux) [inline packages/core/src/components/datetime-input/datetime-input.types.ts:1]: Do not forget to expose the event in the public-api.ts

File: packages/core/src/components/date-time-card/date-time-card.tsx (hunk 2 above)
Comment (danielleroux) [reply packages/core/src/components/date-time-card/date-time-card.tsx:46]: @nuke-ellington Since tag not necessary its an internal component

File: packages/core/src/components/time-picker/time-picker.tsx [hunk 13]
```diff
@@ -540,23 +540,34 @@ export class TimePicker extends Mixin(...DefaultMixins) {
...
+      } else {
+        // We hit a shadow boundary, try to get the host element
+        const rootNode = currentElement.getRootNode();
+        if (rootNode && (rootNode as ShadowRoot).host) {
+          currentElement = (rootNode as ShadowRoot).host as Element;
+        } else {
+          break;
+        }
+      }
+    }
```
Comment (danielleroux) [inline packages/core/src/components/time-picker/time-picker.tsx]: I really dont get it what are you try to solve here? Can you please explain more detail?

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 9 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: removed. thanks

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 11 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: removed. thanks

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 12 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: Both date-input and time-input components use /** @internal */ on the exact same public properties. These properties need to be public (not private) because they're accessed by validation utilities and form frameworks, but the @internal JSDoc tells documentation generators to exclude them from public API docs.

should it be removed?

File: packages/core/src/components/datetime-input/datetime-input.types.ts
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.types.ts:1]: added. thanks

File: packages/core/src/components/time-picker/time-picker.tsx (hunk 13 above)
Comment (alexkaduk) [reply packages/core/src/components/time-picker/time-picker.tsx]: The ix-datetime-picker contains an ix-time-picker, and they're all nested inside an ix-dropdown. With web components using shadow DOM, this traversal fix ensures the time picker can always find its parent dropdown and properly observe when it opens/closes.
If I remove the shadow DOM traversal:
  - ✅ Component renders
  - ✅ Clicking works
  - ❌ Time doesn't scroll to current value when opening
  - ❌ State doesn't reset when closing
  - ❌ Reopening shows wrong time/focus from previous session

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 10 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: updated. thanks

File: packages/core/src/components/time-picker/time-picker.tsx (hunk 13 above)
Comment (danielleroux) [reply packages/core/src/components/time-picker/time-picker.tsx]: Checkout https://github.com/siemens/ix/blob/a35d9b78409a0927193f4277f35b03490dd0d3d4/packages/core/src/components/utils/shadow-dom.ts#L40-L58

which does exactly this one

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 12 above)
Comment (danielleroux) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: Yes because there is no documentation generator for properties without `@Prop/@Event/@Method`

Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: removed. also deleted same for date-input and time-input

File: packages/core/src/components/time-picker/time-picker.tsx (hunk 13 above)
Comment (alexkaduk) [reply packages/core/src/components/time-picker/time-picker.tsx]: refactored setupVisibilityObserver to use closestPassShadow from shadow-dom.ts. thanks!

File: packages/core/src/components/datetime-input/datetime-input.tsx [hunk 14]
```diff
@@ -0,0 +1,678 @@
...
+  }
+
+  private readonly handleDateSelect = (event: CustomEvent) => {
+    const { from, time } = event.detail;
+
+    if (!from || !time) {
+      return;
+    }
+
+    const displayValue = `${from} ${time}`;
```
Comment (danielleroux) [inline packages/core/src/components/datetime-input/datetime-input.tsx]: This look to my like a issue of the picker combining formats. Because you force the user to use a delimiter 
> `' '`

This prevents to have formats like `2026-03-11T10:00:00.000+01:00` (ISO string).

In my opinion it makes no sense to hold time format and date format in two distinct format fields. Because you pick ONE date (date + time) in a single date.

**Review by danielleroux — CHANGES_REQUESTED** [review]
Restore valid datetime is not possible without interacting with with both components date and time:

- Enter some valid date e.g "2026/03/12 08:42:24"
- Change the seconds to 99
- Selecting in the seconds = 30
- Press "Confirm"
-- > Picker still invalid

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 4 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: udpated, now it is like this
```
  /** ARIA label for previous month navigation button */
  @Prop() ariaLabelPreviousMonthButton?: string = 'Previous month';

  /** ARIA label for next month navigation button */
  @Prop() ariaLabelNextMonthButton?: string = 'Next month';
```

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 5 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: updated to this
```
         <ix-icon-button
            aria-hidden={this.ariaLabelCalendarButton}
            tabindex={-1}
            class={{ 'calendar-hidden': this.disabled || this.readonly }}
            variant="subtle-tertiary"
            icon={iconCalendar}
            onClick={(event) => this.onCalendarClick(event)}
          ></ix-icon-button>
```

File: packages/core/src/components/datetime-input/datetime-input.tsx (hunk 14 above)
Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: datetime-input uses datetime-picker that has two props
1) dateFormat
https://ix.siemens.io/docs/components/date-time-picker/code#property-dateFormat
2) timeFormat
https://ix.siemens.io/docs/components/date-time-picker/code#property-timeFormat

if one prop should be used, might it be done for both components with next PR?

Comment (danielleroux) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: No not needed imo both pickers use luxon:
 
e.g
 
`<ix-time-picker timeFormat={this.format}`
 
`<ix-date-picker dateFormat={this.format}`
 
maybe in the future we can align it but its getting only a problem if you combine the both values into a single display value.
This usecase does not hit the datetime-picker because this component dont show any combine date time.

Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: # Suggested datetime-input API proposal: `format` + `timePicker`

## Overview

Replace `dateFormat` and `timeFormat` with a single `format` prop and an optional `timePicker` object.

## API

| Prop         | Type     | Default                 | Purpose                                                                                                        |
| ------------ | -------- | ----------------------- | -------------------------------------------------------------------------------------------------------------- |
| `format`     | `string` | `'yyyy/LL/dd HH:mm:ss'` | Single source of truth for the **value** (parsing, display, validation)                                        |
| `timePicker` | `object` | —                       | Configuration for the **time picker view** (UI behavior and layout). Defaults inherited from `ix-time-picker`. |

### `timePicker` shape

```typescript
timePicker?: {
  format?: string;
  hourInterval?: number;
  minuteInterval?: number;
  secondInterval?: number;
  millisecondInterval?: number;
  hideHeader?: boolean;
  i18nHeader?: string;
  i18nHourColumnHeader?: string;
  i18nMinuteColumnHeader?: string;
  i18nSecondColumnHeader?: string;
  i18nMillisecondColumnHeader?: string;
};
```

All defaults are inherited from `ix-time-picker` (see [time-picker API](https://ix.siemens.io/docs/components/time-picker/code)).

## How they relate

- `format` defines the **value**. It is used for parsing user input, formatting the output value, and validation. No splitting of this format is needed.
- `timePicker.format` defines the **time picker UI** (which columns are shown, 12h vs 24h). It is independent from `format`.
- They **can differ**. For example: `format="yyyy/LL/dd HH:mm:ss"` with `timePicker.format="hh:mm a"` (value in 24h, picker shows 12h).

## Combined value

The value is always parsed and formatted using `format` only.

When the user confirms a selection in the picker:

1. The date picker returns the selected date in its default format (`'yyyy/LL/dd'`).
2. The time picker returns the selected time (in `timePicker.format`).
3. `datetime-input` converts both to a `DateTime` object and formats it with `format` for the final value.

## Usage examples

```html
<!-- Default: value format "yyyy/LL/dd HH:mm:ss", time picker shows h/m/s columns (24h) -->
<ix-datetime-input format="yyyy/LL/dd HH:mm:ss" />

<!-- Custom date format -->
<ix-datetime-input format="dd.MM.yyyy HH:mm" />

<!-- 12h time picker, 24h value -->
<ix-datetime-input
  format="yyyy/LL/dd HH:mm:ss"
  time-picker='{"format": "hh:mm a"}'
/>

<!-- 15-minute intervals -->
<ix-datetime-input
  format="yyyy/LL/dd HH:mm"
  time-picker='{"minuteInterval": 15}'
/>

<!-- Milliseconds -->
<ix-datetime-input
  format="yyyy/LL/dd HH:mm:ss.SSS"
  time-picker='{"format": "HH:mm:ss.SSS"}'
/>
```

## Delimiter

Delimiter is taken from the `format` prop and handled by Luxon natively.

- `format="yyyy/LL/dd HH:mm:ss"` → space delimiter
- `format="yyyy-MM-dd'T'HH:mm:ss"` → `T` as quoted literal delimiter

## minDate / maxDate

`minDate` and `maxDate` follow the `format` prop.

```typescript
/** Minimum allowed date matching the format prop (e.g., "2026/01/20 00:00:00" for format "yyyy/LL/dd HH:mm:ss") */
@Prop() minDate?: string;

/** Maximum allowed date matching the format prop (e.g., "2026/12/31 23:59:59" for format "yyyy/LL/dd HH:mm:ss") */
@Prop() maxDate?: string;
```

```html
<ix-datetime-input
  format="yyyy/LL/dd HH:mm:ss"
  min-date="2026/01/20 00:00:00"
  max-date="2026/12/31 23:59:59"
/>

<ix-datetime-input
  format="dd.MM.yyyy HH:mm"
  min-date="20.01.2026 00:00"
  max-date="31.12.2026 23:59"
/>
```

Internally, `datetime-input` parses minDate/maxDate with `format` and converts them to the date picker's default format (`'yyyy/LL/dd'`) before passing them through.

**Bug fix needed in `datetime-picker`**: `i18nTime` prop exists but is never forwarded to `ix-time-picker` as `i18nHeader`. This should be fixed so `timePicker.i18nHeader` works as expected.

## Prop changes summary

| Current prop | Action | Replacement |
|-------------|--------|-------------|
| `dateFormat` | Removed | `format` (full datetime format) |
| `timeFormat` | Moved to `timePicker` | `timePicker.format` |
| `i18nTime` | Moved to `timePicker` | `timePicker.i18nHeader` |

## Implementation notes

- `dateFormat` and `timeFormat` are removed (replaced by `format` and `timePicker`).
- `timePicker` options are passed from `datetime-input` → `ix-datetime-picker` → `ix-time-picker`. Since `ix-time-picker` is rendered inside `ix-datetime-picker`, `datetime-picker` needs to be updated to accept and forward these options.
- **`ix-datetime-picker` changes required** — currently only `format` (`timeFormat`) is forwarded. The following props need to be added to `ix-datetime-picker` and forwarded to `ix-time-picker`:
  - `hourInterval`
  - `minuteInterval`
  - `secondInterval`
  - `millisecondInterval`
  - `hideHeader`
  - `i18nHeader` (also fixes the existing dead `i18nTime` prop)
  - `i18nHourColumnHeader`
  - `i18nMinuteColumnHeader`
  - `i18nSecondColumnHeader`
  - `i18nMillisecondColumnHeader`
- `datetime-input` converts all values internally:
  - Date values → date picker's default format (`'yyyy/LL/dd'`)
  - Time values → `timePicker.format` (default `'TT'`)
  - Combined value → `format`

Comment (danielleroux) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: tldr :) I will check the code not the ai comment

Comment (alexkaduk) [reply packages/core/src/components/datetime-input/datetime-input.tsx]: only `format` prop is used
